### PR TITLE
Refactor the Aes encryption implementation

### PR DESCRIPTION
**Pull request by eerhardt**
Refactor the Aes encryption implementation to reuse the same design as the TripleDes implementation.

This cuts down on a lot of duplicate code, since we can reuse the UniversalCryptoTransform classes.

@bartonjs @stephentoub

### CODE REVIEW

File: src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs [hunk 1]
```diff
@@ -1,34 +1,74 @@
...
+                    foundKeysize = true;
+                }
+            }
+
+            if (!foundKeysize)
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+            }
+
+            // This is what AesCngCryptoTransform::SetCipherMode throws when it doesn't understand the value.
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs]: Presumably you've deleted that type, so the comment is now out of date :smile:.

Comment (eerhardt) [inline src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs]: Yep, missed that one.

**Comment by bartonjs** [comment]
Aside from the comment, LGTM.

@AtsushiKan any other thoughts?

File: src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs [hunk 2]
```diff
@@ -1,34 +1,74 @@
...
-        private static ICryptoTransform CreateDecryptor(
-            CipherMode cipherMode,
-            PaddingMode paddingMode,
-            byte[] key,
-            byte[] iv,
-            int blockSize)
-        {
-            return new AesOpenSslCryptoTransform(cipherMode, paddingMode, key, iv, blockSize, false);
+            // The algorithm pointer is a static pointer, so not having any cleanup code is correct.
+            IntPtr algorithm = algorithmFunc();
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs]: Is there any other caller of FindAlgorithmSelector that needs to do something other than invoke it?  I'm wondering if FindAlgorithmSelector should just be SelectAlgorithm and effectively include this delegate invocation.

**Comment by stephentoub** [comment]
One nit, otherwise LGTM.

**Comment by AtsushiKan** [comment]
LGTM

File: src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs (hunk 2 above)
Comment (eerhardt) [inline src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AesImplementation.Unix.cs]: No, there isn't another caller. I was just porting code straight over as it was.

I'll make this change.

**Comment by eerhardt** [comment]
@dotnet-bot test this please

**Comment by eerhardt** [comment]
@dotnet-bot test this please

**Pull request merged**